Model/Analytics.php

```diff
@@ -55,84 +49,21 @@ public function addPageEvents()
         if (!$this->fullActionName || $this->_isRejected($this->fullActionName)) {
             return;
         }
-
-        // Catalog search pages
-        if ($this->fullActionName == 'catalogsearch_result_index') {
-            $query = $this->_searchHelper->getEscapedQueryText();
-            if ($query) {
-                $params = ['query' => $query];
-                $this->addEvent('track', 'search', $params);
-                return;
-            }
-        }
-
-        // category view pages
-        if ($this->fullActionName == 'catalog_category_view') {
-            $category = $this->_coreRegistry->registry('current_category');
-
-            $data =  [
-                'id'    =>  $category->getId(),
-                'name'  =>  $category->getName()
-            ];
-
-            $this->addEvent('track', 'view_category', $data);
-            return;
-        }
-
-        // product view pages
-        if ($this->fullActionName == 'catalog_product_view') {
-            /** @var \Magento\Catalog\Model\Product $product */
-            $product = $this->_coreRegistry->registry('current_product');
-            $data =  [
-                'id'    => $product->getId(),
-                'sku'   => $product->getSku(),
-                'name'  => $product->getName(),
-                'price' => $product->getPrice(),
-                'url'   => $product->getProductUrl()
+        
+        switch($this->fullActionName) {
+            // product view pages
+            case 'catalog_product_view':
+                /** @var \Magento\Catalog\Model\Product $product */
```

Comment:
Here:
```php
event = ProductViewEvent.new()
$this->addEvent('track', 'view_product', $data);
```
ProductViewEvent.php
```php
public function __construct(
    \Magento\Framework\Registry $registry
) {
    $this->_coreRegistry = $registry;
}
public function callJS() {
    $product = $this->_coreRegistry->registry('current_product');
    return "window.metrilo.viewProduct(product.id);";
}
```

Reply:
Removed Model/Analytics and moved event logic to the Block/Analytics class.